nanotune/device_tuner/dottuner.py

```diff
@@ -1,581 +1,1027 @@
-                )
-            )
-            segment_info = tuningresult.features["segment_info"]
-            if desired_regime in segment_info[:, 2]:
-                logger.info("Desired regime found.")
-                done = True
-
-            if done and take_high_res:
-
-                logger.info("Take high resolution of entire charge diagram.")
-                tuningresult = self.get_charge_diagram(
+            if tuningresult.success and take_high_res:
+                self.take_high_resolution_diagram(
                     device,
-                    plungers,
-                    voltage_precision=0.0005,
-                )
-                self._tuningresults_all[device.name].add_result(
-                    tuningresult, f"chargediagram_highres_{n_iter}"
+                    device_layout.plungers(),
+                    take_segments=True,
                 )
-                logger.info("Take high resolution of charge diagram segments.")
-                segment_info = tuningresult.features["segment_info"]
-                for sid, segment in enumerate(segment_info):
-                    if segment[2] == desired_regime:
-                        readout_meth = segment[1].keys()[0]
-                        v_ranges = segment[1][readout_meth]
-                        axis_keys = ["range_x", "range_y"]
-                        for plunger, key in zip(plungers, axis_keys):
-                            plunger.current_valid_range(v_ranges[key])
-
-                        tuningresult = self.get_charge_diagram(
-                            device,
-                            plungers,
-                            voltage_precision=0.0001,
-                        )
-                        self._tuningresults_all[device.name].add_result(
-                            tuningresult, f"chargediagram_{n_iter}_segment_{sid}"
-                        )
-
+            done = tuningresult.success
+            success = tuningresult.success
             if continue_tuning:
                 done = False
                 logger.warning("Continue tuning regardless of the outcome.")
 
             if not done:
                 self.update_gate_configuration(
-                    device,
-                    tuningresult,
-                    desired_regime,
-                )
-                logger.info(
-                    (
-                        "Continuing with new configuration: "
-                        f"{device.get_gate_status()}."
-                    )
+                    device, tuningresult, target_state,
                 )
 
         if n_iter >= max_iter:
-            logger.info(f"Tuning {device.name}: Max number of iterations reached.")
-
-        return self._tuningresults_all[device.name]
+            logger.info(
+                f"Tuning {device.name}: Max number of iterations" \
+                f"reached.")
+        return success
 
-    def update_top_barrier(
+    def take_high_resolution_diagram(
```

Comment:
Should some of these methods be private if they're not being called externally?

Reply:
I could imagine to use this method outside of the tuning procedure, when tuning half manually for example. It happened to me to interrupt an automated tuning to take high res data if I needed more data for (potential) papers.